### ISSUE TITLE
Django querysets default to unfiltered when empty

### ISSUE THREAD

**Issue by defrex** [graphql-python/graphene#116]
When using a `DjangoConnectionField` and performing a filter on the queryset in the resolve method, an empty queryset will result all the objects being returned.

For example

```python
import graphene
from graphene.contrib.django.fields import DjangoConnectionField
from graphene.contrib.django.types import DjangoNode

from .models import Category


class CategoryNode(DjangoNode):
    class Meta:
        model = Category


class Query(graphene.ObjectType):
    categories = DjangoConnectionField(
        CategoryNode,
        name=graphene.String(),
    )

    def resolve_categories(self, args, info):
        return Category.objects.filter(name__icontains=args['name'])
```

```graphql
query {
  hasMatch: categories(name: "food") {
    edges {
      node {
        name
      }
    }
  }
  noMatch: categories(name: "no matches asdf") {
    edges {
      node {
        name
      }
    }
  }
}
```

```js
{
  "data": {
    "hasMatch": {
      "edges": [
        [...just "food" categories returned]
      ]
    },
    "noMatch": {
      "edges": [
        [...all categories returned]
      ]
    }
  }
}
```

In this simple case. If the `name` arg returns a queryset with results, those results will be used. Otherwise, all categories will be returned.

This doesn't apply to `django-filters` based filtering, just these simple cases. However, if you're using these simple filters for something like security then this becomes a huge issue.

**Comment by syrusakbary** [comment]
Which graphene version are you using?

**Comment by defrex** [comment]
Looks like this is solved as of `0.7.2`. I upgraded yesterday afternoon, and I can no longer reproduce.

Keep up the great work!

**Comment by syrusakbary** [comment]
Thanks! 
This commit fixed the issue: https://github.com/graphql-python/graphene/commit/314703d7b5380c80fe2ee1e2ad4a3eb3d9ed7fc4#diff-233528cf66f4e9df736a367fe12b2ec1R30
